modules/standard/Reflection.chpl

```diff
@@ -44,6 +44,10 @@ private proc checkQueryT(type t) type {
   else
     compilerError(t:string, " is not a class, record, or union type", 2);
 }
+private proc checkValidQueryT(type t) param {
+  if !isClassType(t) && !(isRecordType(t) || isUnionType(t)) then
```

Comment:
Mixing `&&` and `||` in this way seems a little unnecessarily complicated, but it seems to be accomplishing what I'd expect so it's probably fine to leave it.